spkrepo/templates/frontend/package.html

```diff
@@ -14,19 +13,38 @@ <h1 class="media-heading">{{ package.versions[-1].displaynames['enu'].displaynam
         </small>
       </h1>
       <p>{{ package.versions[-1].descriptions['enu'].description }}</p>
+      <div class="package-screenshots">
+        {% for screenshot in package.screenshots %}
+        <img class="package-screenshot" src="{{ url_for('nas.data', path=screenshot.path) }}"/>&nbsp;
+        {% endfor %}
+      </div>
+      {% for version in package.versions|reverse %}
       <dl>
+        <dt>Version {{ version.version_string | safe }}</dt>
+        <dd>{{ version.changelog | safe }}</dd>
+        <dt>Date</dt>
+        <dd>{{ version.insert_date }}</dd>
         <dt>Architectures</dt>
         <dd>
-          {% for build in package.versions[-1].builds %}
+          {% for (version, builds) in version.builds_per_dsm.items() %}
+          <!-- Group firmware by DSM or SRM if 1.x -->
+          {% if version == '1' %}
+          SRM {{ version }}.x:
+          {% else %}
+          DSM {{ version }}.x:
+          {% endif %}
+          {% for build in builds %}
           {% for arch in build.architectures %}
-          <span class="label label-default">{{ arch.code }}</span>
+          <a href="{{ url_for('nas.data', path=build.path) }}"><span class="label label-default">{{ build.firmware.version }} {{ arch.code }}</span></a>
           {% endfor %}
           {% endfor %}
+          <br/>
+          {% endfor %}
         </dd>
-        <dt>Author</dt>
-        <dd>{{ package.author.username }}</dd>
```

Comment:
The test is complaining about the missing author. Fine by me if you want to get rid of it but your need to adjust the test case to reflect that.

Reply:
OK. I do not remember why I decided to remove it... I will add it back

Reply:
Now I remember... "author" in spkrepo is the user who has published the package, not the real package "maintainer"... now we work in team with many contributors, it would be fair to display only package maintainer